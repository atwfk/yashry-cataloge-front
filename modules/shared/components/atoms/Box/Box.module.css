.box {
  @apply shadow-soft pb-4 overflow-hidden rounded-md flex flex-col items-center space-y-1;
}
