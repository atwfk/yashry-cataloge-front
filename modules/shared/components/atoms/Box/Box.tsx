import React from "react";
import type { FC, ReactElement } from "react";
import styles from "./Box.module.css";

const Box: FC = ({ children }): ReactElement => (
  <div className={styles.box}>{children}</div>
);

export default Box;
