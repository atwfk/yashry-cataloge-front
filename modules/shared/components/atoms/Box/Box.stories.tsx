import type { ReactElement } from "react";
import type { Story } from "@storybook/react";
import Box from "./Box";

const args = {
  title: "Atoms/Box",
  component: Box,
};

export default args;

const Template: Story = (args): ReactElement => (
  <Box {...args}>normal text</Box>
);

export const box = Template.bind({});
box.args = {};
