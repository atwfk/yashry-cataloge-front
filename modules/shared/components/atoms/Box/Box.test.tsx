import React from "react";
import Box from "./Box";
import renderer from "react-test-renderer";

describe("Box", () => {
  it("should render Box component with padding and shadow", () => {
    const tree = renderer.create(<Box>normal Text</Box>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
