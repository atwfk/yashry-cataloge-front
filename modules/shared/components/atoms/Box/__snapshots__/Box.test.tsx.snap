// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box should render Box component with padding and shadow 1`] = `
<div
  className="box"
>
  normal Text
</div>
`;
